yang-module-versioning/draft-ietf-netmod-yang-module-versioning.xml

```diff
@@ -1703,17 +1702,11 @@ module ietf-yang-library-revisions {
                <t>A new schema node, e.g., "vpn-name", of type string is added to the same location as the existing node
                "vpn-id". This new node has status "current" and its description explains that it is replacing node
                "vpn-id".</t>
-               <t>During the period of time when both schema nodes are supported, the interactions between the two nodes is outside the scope of this document and will vary on a case by case basis. Here are some options:
-               <list style="numbers">
-                  <t>A server may prevent the new node from being set if the old node is already set (and
-                  vice-versa). A "choice" construction could be used, or the new node may have a "when" statement to achieve this. The old node must not have a "when"
-                  statement since this would be an NBC change, but the server could reject the old
-                  node from being set if the new node is already set.</t>
-                  <t>If the new node is set and a client does a get or get-config operation on the old node, the server
-                  could map the value. For example, if the new node "vpn-name" has value "123" then the server could return
-                  integer value 123 for the old node "vpn-id". However, if the value can not be mapped then the configuration
-                  would be incomplete. The behavior in this case is outside the scope of this document.</t>
-               </list>
+               <t>During the period of time when both schema nodes are supported, the interactions between the two nodes is outside the scope of this document and will vary on a case by case basis. One possible option is to have the 
+               server  prevent the new node from being set if the old node is already set (and
+               vice-versa). The new node could have a "when" statement to achieve this. The old node must not have a "when"
```

Comment:
Maybe, "one and only one of the nodes could have a 'when' statement to achieve this.  Both nodes cannot have 'when' statements since this would be an NBC change overall."

Reply:
Cascade this down to other places...

Reply:
I'm not sure that either (XOR) of the nodes (new or old) can have a "when". We can't add a "when" to the old node without that being an NBC change. Theoretically you could add a "when" on the old node, as long as the "when" only depends on the existence of the new node (which an old client would never set), but that is pretty subtle and breaks the generic rule that you can't add a new "when" statement in 7950. So I think we might need to keep wording that says you can add a "when" to the *new* node, but not the old one.

Reply:
That's fair.  I just felt it wasn't so clear when I read it.  Maybe:

>The new node could have a 'when' statement added to it to achieve this.  However, the old node cannot have a 'when' statement added as this would be an NBC change.

Reply:
Ahh yes - I like that. I wasn't clear that it can't have a when *added* (or modified). But if it already had a when that's fine.

How about this:
The new node could have a "when" statement added to it to achieve this. The old node, however, must not have a "when" statement added, or an existing "when" modified to be more restrictive, since this would be an NBC change. In any case, the server could reject the old node from being set if the new node is already set.